Sources/joyfill-components-swift/Components/MultipleChoice/MultipleChoice.swift

```diff
@@ -0,0 +1,222 @@
+            titleLabel.heightAnchor.constraint(equalToConstant: 50),
+            
+            // TableView Constraints
+            tableView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: -8),
+            tableView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+        ])
+        
+        // tableView Properties
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.layer.borderWidth = 1
+        tableView.layer.cornerRadius = 10
+        tableView.separatorStyle = .none
+        tableView.contentInsetAdjustmentBehavior = .never
+        tableView.showsVerticalScrollIndicator = false
+        tableView.layer.borderColor = UIColor(hexString: "#C0C1C6")?.cgColor
+        tableView.register(MultiChoiceTableViewCell.self, forCellReuseIdentifier: "MultiChoiceTableViewCell")
+        
+        titleLabel.borderWidth = 0
+        titleLabel.textColor = .black
+        titleLabel.font = UIFont.systemFont(ofSize: 14, weight: .semibold)
+    }
+}
+
+// MARK: Setup tableView
+extension MultipleChoice: UITableViewDelegate, UITableViewDataSource {
+    // MARK: TableView delegate method for number of rows in section
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return choiceOptionsArray.count
+    }
+    
+    // MARK: TableView delegate method for cell for row at
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "MultiChoiceTableViewCell", for: indexPath as IndexPath) as! MultiChoiceTableViewCell
+        if multiSelect == true {
+            if selectedIndexPath.contains(indexPath.row) {
+                cell.myCheckbox.isChecked = true
+                cell.myCheckbox.checkboxFillColor = UIColor(hexString: "#3767ED") ?? .lightGray
+            } else {
+                cell.myCheckbox.isChecked = false
+                cell.myCheckbox.checkboxFillColor = .white
+            }
+        } else {
+            if selectArray == indexPath.row {
+                cell.myCheckbox.isChecked = true
+                cell.myCheckbox.checkboxFillColor = UIColor(hexString: "#3767ED") ?? .lightGray
+            } else {
+                cell.myCheckbox.isChecked = false
+                cell.myCheckbox.checkboxFillColor = .white
+            }
+        }
+        cell.selectionStyle = .none
+        cell.myCheckbox.isUserInteractionEnabled = false
+        cell.cellLabel.text = choiceOptionsArray[indexPath.row]
+        return cell
+    }
+    
+    // MARK: TableView delegate function called when cell is selected
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if multipleChoiceDspMode == "readonly" {
```

Comment:
This check should be the code below as there is no point having an empty true state of an if statement

```
if multipleChoiceDspMode !== "readonly" {
  checkForMultiSelectOrSingleSelect(at: indexPath)
}
```